### PR TITLE
Google Roads API. Nearest Roads support with tests

### PR DESCRIPTION
**Pull request by vanitu**
Provide NearestRoads support to Google Roads API.
Include spec tests.

### CODE REVIEW

**Comment by vanitu** [comment]
Hi, this is my first contributing attempt. Pls advise if something missed or done wrogly.
Thanks.

**Comment by edwardsamuel** [comment]
Hi @vanitu, thanks for the pull request. I think your commit is based on `master` branch and it would be better if you merge this to `master` instead. So later, I can merge it and publish this gem easier. And also, the `develop` branch is currently under heavy changes, which may more hassle to be merged.

File: lib/google_maps_service/apis/roads.rb [hunk 1]
```diff
@@ -104,6 +104,42 @@ def snapped_speed_limits(path)
                  custom_response_decoder: method(:extract_roads_body))
     end
 
+    # Returns the Nearest road segments for provided points in a path.
```
Comment (edwardsamuel) [inline lib/google_maps_service/apis/roads.rb]: No need capitalize on `Nearest` :)

File: lib/google_maps_service/apis/roads.rb [hunk 2]
```diff
@@ -104,6 +104,42 @@ def snapped_speed_limits(path)
...
+    #
+    # @param [Array] path The path to be match to nearest road segment. Array of latitude/longitude pairs.
+    #
+    # @return [Array] Array of snapped points.
+
+    def nearest_roads(points)
+      points = GoogleMapsService::Convert.waypoints(points)
+
+      params = {
+              points: points
```
Comment (edwardsamuel) [inline lib/google_maps_service/apis/roads.rb]: Please use 2 space indentation to make it consistent across the project.

File: spec/google_maps_service/apis/roads_spec.rb [hunk 3]
```diff
@@ -17,6 +17,20 @@
     end
   end
 
+  context '#nearest_roads' do
+    before(:example) do
+      stub_request(:get, /https:\/\/roads.googleapis.com\/v1\/nearestRoads/)
+              .to_return(:status => 200, headers: { 'Content-Type' => 'application/json' }, body: '{"snappedPoints":["foo"]}')
```
Comment (edwardsamuel) [inline spec/google_maps_service/apis/roads_spec.rb]: Same as before, please use 2 space indentation.

File: lib/google_maps_service/apis/roads.rb [hunk 4]
```diff
@@ -104,6 +104,42 @@ def snapped_speed_limits(path)
...
+    # @example Multi points snap
+    #   path = [
+    #       [-33.8671, 151.20714],
+    #       [-33.86708, 151.20683000000002],
+    #       [-33.867070000000005, 151.20674000000002],
+    #       [-33.86703, 151.20625]
+    #   ]
+    #   results = client.nearest_roads(path)
+    #
+    # @param [Array] path The path to be match to nearest road segment. Array of latitude/longitude pairs.
```
Comment (edwardsamuel) [inline lib/google_maps_service/apis/roads.rb]: The param name should be `points` as in the method declaration. 

I think "match" sounds not really fit here, any other words?
Or, you can also move:

```
    # Takes up to 100 independent coordinates, and returns the closest road segment for each point.
    # The points passed do not need to be part of a continuous path.
```

to `points` description.

File: lib/google_maps_service/apis/roads.rb [hunk 5]
```diff
@@ -104,6 +104,42 @@ def snapped_speed_limits(path)
...
+    # Returns the Nearest road segments for provided points in a path.
+    #
+    # Takes up to 100 independent coordinates, and returns the closest road segment for each point.
+    # The points passed do not need to be part of a continuous path.
+    #
+    # @example Single point snap
+    #   results = client.nearest_roads([40.714728, -73.998672])
+    #
+    # @example Multi points snap
+    #   path = [
```
Comment (edwardsamuel) [inline lib/google_maps_service/apis/roads.rb]: Use `points` as variable name. See next comment.

File: spec/google_maps_service/apis/roads_spec.rb [hunk 6]
```diff
@@ -17,6 +17,20 @@
...
+    end
+
+    context 'match' do
```
Comment (edwardsamuel) [inline spec/google_maps_service/apis/roads_spec.rb]: Umm, `match` is also not suitable as context name here, cmiiw. I think the context name can be `single point`.

**Comment by vanitu** [comment]
Pull request moved to master

**Pull request closed without merging**